enterprise/server/githubapp/githubapp.go

```diff
@@ -586,6 +592,98 @@ func (a *GitHubApp) GetAccessibleGitHubRepos(ctx context.Context, req *ghpb.GetA
 	return &ghpb.GetAccessibleReposResponse{RepoUrls: urls}, nil
 }
 
+func (a *GitHubApp) CreateRepo(ctx context.Context, req *repb.CreateRepoRequest) (*repb.CreateRepoResponse, error) {
+	tu, err := a.env.GetUserDB().GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if tu.GithubToken == "" {
+		return nil, status.UnauthenticatedError("github account link is required")
+	}
+
+	// Pick the right client based on the request (organization or user).
+	githubClient, token, err := a.newAuthenticatedClient(ctx, tu.GithubToken)
+	if req.Organization != "" {
+		githubClient, token, err = a.newInstallationClient(ctx, req.InstallationId)
+	}
```

Comment:
you're shadowing the first err here with the second before checking the first -- is this intentional?

Reply:
yeah, i don't really care about that original error if we're supposed to be using an installationClient

could make this an if/else but this allows me to avoid having to create all 3 variables before the if else.

down to do this some other way if you have a better idea

Reply:
Done